mypy/build.py

```diff
@@ -544,12 +544,20 @@ def log(self, message: str) -> None:
 def remove_cwd_prefix_from_path(p: str) -> str:
     """Remove current working directory prefix from p, if present.
 
+    Also crawl up until a directory without __init__.py is found.
```

Comment:
It looks like this logic shows up in two places now, with slightly different semantics -- here and in `mypy.main.crawl_up`. Do we need both of them? Seems like we risk confusion or future bugs by having two versions to choose from.